core/thread/src/TRWMutexImp.cxx

```diff
@@ -113,5 +116,11 @@ template class TRWMutexImp<ROOT::TSpinMutex>;
 template class TRWMutexImp<std::mutex>;
 template class TRWMutexImp<TMutex, ROOT::Internal::UniqueLockRecurseCount>;
 template class TRWMutexImp<ROOT::TSpinMutex, ROOT::Internal::UniqueLockRecurseCount>;
-
-} // End of namespace ROOT
\ No newline at end of file
+#if __cplusplus >= 201402L
+template class TRWMutexImp<std::shared_timed_mutex, ROOT::Internal::RecurseCountsShared>;
```

Comment:
What's the advantage of using the `shared_timed_mutex` in this context?

Reply:
The regular ```shared_mutex``` is only available from C++17, so the ```shared_timed_mutex``` is a fallback for the C++14 case.  (I'm not sure whether the additional timed functionality actually makes it slower if not used.)

Reply:
Can you publish the numerical result of the test benchmark?  (for a C++14 and C++17 builds)